src/Build/Task/Typed.hs

```diff
@@ -7,9 +7,11 @@ module Build.Task.Typed (TTask, Key (..), digits, fetch) where
 import Data.Functor.Const
 import Data.Functor.Identity
 
-type family Value k :: *
+class KeyC k where
+  type Value k :: *
+  showKey :: k -> String
```

Comment:
It may be useful to have a way to extract the list of "atomic key", which in the example below will be `[Int]`. I'll play with this idea.